internal/store/device.go

```diff
@@ -185,9 +193,11 @@ func (s *DeviceStore) Get(ctx context.Context, orgId uuid.UUID, name string) (*a
 
 func (s *DeviceStore) createDevice(device *model.Device) (bool, error) {
 	device.Generation = lo.ToPtr[int64](1)
+	originalResourceVersion := device.ResourceVersion
 	device.ResourceVersion = lo.ToPtr[int64](1)
 	if result := s.db.Create(device); result.Error != nil {
 		err := flterrors.ErrorFromGormError(result.Error)
+		device.ResourceVersion = originalResourceVersion
```

Comment:
I don't understand why is this needed.  We create the device from the `api.Device` during every retry iteration.

Reply:
You're right.  I was fixing the tests and I guess some failure led to me thinking this was an issue.  Reverted.